docs/spec/spec-proposals/f1-fee-distribution/f1_fee_distr.tex

```diff
@@ -85,36 +105,61 @@ \subsection{Slashing}
 
-Make each block have an inflation number, by which every staked token should produce $x$ additional staking tokens. In state there is a variable for the sum of all such inflation numbers. Then each period will store this total inflation sum in addition to $\sum_{i=0}^{end}\frac{T_i}{n_i}$. When withdrawing perform a subtraction on the inflation sums at the end and at the start to see how many new staking tokens to produce per staked token. 
+In state a variable should be kept for the sum of all $inflation$ evaluations from now until genesis.
+Then each period will store this total inflation sum in addition to what it already stores per-period.
+The presence of comission rates causes the distribution of delegated stake within a validator to change per block.
+To get around this, when accumulating the value per validator in state, instead of calculating the rewards you'd get for 1 stake, you treat it as though you have $\prod_0^{now} 1 + x$ stake, where $\prod_0^{now} 1 + x$ is the amount of stake accumulated since genesis until now if you had $1$ stake originally bonded.
+When withdrawing, you take the difference as before, which yields the amount of rewards you would have obtained with $(\prod_0^{begin\ bonding\ period}1 + x)$ stake from the block you began bonding at until now. You then divide that by $(\prod_0^{begin\ bonding\ period}1 + x)$ to normalize it, and then proceed as before.
 
-This works great in the model where the inflation rate should be dynamic each block, but apply the same to each validator. Inflation creation can trivially be epoched as long as inflation isn't required within the epoch, through changes to the $x$ function.
+TODO: Explain this better / put equations to prove correctness
+
+This works great in the model where the inflation rate should be dynamic each block, but apply the same to each validator.
+Inflation creation can trivially be epoched as long as inflation isn't required within the epoch, through changes to the $inflation$ function.
 
 Note that this process is extremely efficient.
 
-The above can be trivially amended if we want inflation to proceed differently for different validators each block. (e.g. depending on who was offline) It can also be made to be easily adapted in a live chain. It is unclear if either of these two are more desirable settings.
+The above can be trivially amended if we want inflation to proceed differently for different validators each block.
+(e.g. depending on who was offline) It can also be made to be easily adapted in a live chain. (i.e. via governance) 
+It is unclear if either of these two are more desirable settings.
+Note that this inflation model is more fair than the currently implemented inflation algorithm, as this has inflation reward delegators not just validators.
```

Comment:
That is already the case, how so do you mean (that inflation rewards validators, not delegators)? Presently neither automatically re-bonds.

Notably, this solution does not allow individual delegators to choose whether or not they want to re-bond their inflation. Possibly the choice could be made differently by different validators, and then delegators could select validators accordingly.

Reply:
The current inflation mechanism is via block reward, which only goes to the validators, not to the delegators. (Last time I checked, could be wrong)

Its a pretty easy amendment to make the inflation rewards toggle-able with a 2x space increase. I don't really think it makes sense to make that toggle-able under the framing I described here. Fee rewards would make sense to be toggle-able though.

Reply:
There is some amount of inflation per block, yes (I presume that's what you mean by "block reward"). A bit of that is distributed to the proposer, most to all of the validators by stake - but the "validator operators" themselves only receive the amount they charge in commission as a fraction of the validator rewards, the rest of the validator rewards go to delegators proportional by their stake.

I don't think toggleable inflation rewards is really necessary either.

Reply:
Oh cool, didn't realize that was the case. So we could treat that as part of the fee rewards, and then we just have to handle auto-bonding the staking token portion of the fee reward. That would work exactly the same as its described here, so nbd

Reply:
I actually think inflation going through the normal fee process where it gets divied based on number of signatures, with a slight advantage to the proposer is sub-ideal. Instead I think we should have the inflation model be what I describe in here.

Reply:
Hmm, if we don't pay the proposer if there aren't any fees, there's no incentive to propose a block, right?

Reply:
No fees only occurs when there's only txs produced by the proposer. Having no block here is fine (in fact in some sense it is better) as that avoids future syncing overhead/  state bloat.

Reply:
interesting, I think so too, we should update the rest of our software to actually implement this